test/unit/math/prim/mat/prob/std_binormal_lcdf_test.cpp

```diff
@@ -0,0 +1,493 @@
+#include <stan/math/prim/mat.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+#include <iostream>
+
+using Eigen::Dynamic;
+using Eigen::Matrix;
+using std::vector;
+
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y,
+                       vector<T>& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+  for (int i = cntr; i < inp_vec.size(); ++i) {
+    rho.push_back(inp_vec(i));
+  }
+}
+
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y, T& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
```

Comment:
At the very least, these repeated pieces need to be turned into functions.  

Does the `y` need to be passed in here as an argument since it's just being constructed? How about returning it rather than passing it in?